tests/ensemble_evaluator/test_prefect_ensemble.py

```diff
@@ -456,6 +575,74 @@ def test_run_prefect_ensemble(unused_tcp_port, coefficients):
 
 
+@pytest.mark.timeout(60)
+def test_run_prefect_for_function_defined_outside_py_environment(
+    unused_tcp_port, coefficients, tmpdir, function_config
+):
+    with tmpdir.as_cwd():
+        # Create temporary module that defines a function `bar`
+        # 'bar' returns a call to different function 'internal_call' defined in the same python file
+        module_path = Path(tmpdir) / "foo"
+        module_path.mkdir()
+        init_file = module_path / "__init__.py"
+        init_file.touch()
+        file_path = module_path / "bar.py"
+        file_path.write_text(
+            "def bar(coeffs):\n    return internal_call(coeffs)\n"
+            "def internal_call(coeffs):\n    return [sum(coeffs.values())]\n"
+        )
+        spec = importlib.util.spec_from_file_location("foo", str(file_path))
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+        func = getattr(module, "bar")
+        pickle_func = cloudpickle.dumps(func)
+        init_file.unlink()
+        file_path.unlink()
+
+        # Check module is not in the python environment
+        with pytest.raises(ModuleNotFoundError):
+            import foo.bar
+
+        test_path = Path(SOURCE_DIR) / "test-data/local/prefect_test_case"
+        config = function_config
+        coeffs_trans = coefficient_transmitters(
+            coefficients, config.get(ids.STORAGE)["storage_path"]
+        )
+
+        config["realizations"] = len(coefficients)
+        config["executor"] = "local"
+        config["steps"][0]["jobs"][0]["executable"] = pickle_func
+        config["inputs"] = {iens: coeffs_trans[iens] for iens in range(2)}
+        config["outputs"] = output_transmitters(config)
+
+        service_config = EvaluatorServerConfig(unused_tcp_port)
+        config["dispatch_uri"] = service_config.dispatch_uri
+        ensemble = PrefectEnsemble(config)
+        evaluator = EnsembleEvaluator(ensemble, service_config, 0, ee_id="1")
+        res = None
+        with evaluator.run() as mon:
+            for event in mon.track():
+                if event.data is not None and event.data.get("status") in [
+                    "Failed",
+                    "Stopped",
+                ]:
+                    results = mon.get_result()
+                    mon.signal_done()
+        assert evaluator._snapshot.get_status() == "Stopped"
+        successful_realizations = evaluator._snapshot.get_successful_realizations()
+        assert successful_realizations == config["realizations"]
+        transmitted_record = lambda record: asyncio.get_event_loop().run_until_complete(
+            record.load()
+        )
+        expected_results = [
+            pickle.loads(pickle_func)(coeffs) for coeffs in coefficients
+        ]
+        results = [
+            transmitted_record(res["function_output"]).data for res in results.values()
+        ]
+        assert expected_results == results
```

Comment:
Suggesting a approach slightly more consistent with other `Record.load` consumer code.

```suggestion
        transmitter_futures = [
            res["function_output"].load() for res in results.values()
        ]
        results = asyncio.get_event_loop().run_until_complete(
            asyncio.gather(*transmitter_futures)
        )
        expected_results = [
            pickle.loads(pickle_func)(coeffs) for coeffs in coefficients
        ]
        assert expected_results == [res.data for res in results]
```